File: js_sdk/src/model/Oauth2AccessTokenBody.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';

/**
 * The Oauth2AccessTokenBody model module.
 * @module model/Oauth2AccessTokenBody
 * @version 0.1.1
 */
export class Oauth2AccessTokenBody {
  /**
   * Constructs a new <code>Oauth2AccessTokenBody</code>.
   * @alias module:model/Oauth2AccessTokenBody
   * @class
   * @param app_id {String} The App ID applied by the developer. It can be found in the Basic Information section for you app under [My Apps](https://ads.tiktok.com/marketing_api/apps/).
   * @param auth_code {String} Authorization code provided once the callback is complete.
   * @param secret {String} The private key of the developer's App. It can be found in the Basic Information section for your app under [My Apps](https://ads.tiktok.com/marketing_api/apps/).
   */
  constructor(app_id, auth_code, secret) {
    this.app_id = app_id;
    this.auth_code = auth_code;
    this.secret = secret;
  }

  /**
   * Constructs a <code>Oauth2AccessTokenBody</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Oauth2AccessTokenBody} obj Optional instance to populate.
   * @return {module:model/Oauth2AccessTokenBody} The populated <code>Oauth2AccessTokenBody</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new Oauth2AccessTokenBody();
      if (data.hasOwnProperty('app_id'))
        obj.app_id = ApiClient.convertToType(data['app_id'], 'String');
      if (data.hasOwnProperty('auth_code'))
        obj.auth_code = ApiClient.convertToType(data['auth_code'], 'String');
      if (data.hasOwnProperty('secret'))
        obj.secret = ApiClient.convertToType(data['secret'], 'String');
    }
    return obj;
  }
}

/**
 * The App ID applied by the developer. It can be found in the Basic Information section for you app under [My Apps](https://ads.tiktok.com/marketing_api/apps/).
 * @member {String} app_id
 */
Oauth2AccessTokenBody.prototype.app_id = undefined;

/**
 * Authorization code provided once the callback is complete.
 * @member {String} auth_code
 */
Oauth2AccessTokenBody.prototype.auth_code = undefined;

/**
 * The private key of the developer's App. It can be found in the Basic Information section for your app under [My Apps](https://ads.tiktok.com/marketing_api/apps/).
 * @member {String} secret
 */
Oauth2AccessTokenBody.prototype.secret = undefined;

File: java_sdk/src/main/java/business_api_client/Oauth2AccessTokenBody.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
/**
 * Oauth2AccessTokenBody
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-01-27T17:23:30.114869-08:00[America/Los_Angeles]")
public class Oauth2AccessTokenBody {
  @JsonProperty("app_id")
  private String appId = null;

  @JsonProperty("auth_code")
  private String authCode = null;

  @JsonProperty("secret")
  private String secret = null;

  public Oauth2AccessTokenBody appId(String appId) {
    this.appId = appId;
    return this;
  }

   /**
   * The App ID applied by the developer. It can be found in the Basic Information section for you app under [My Apps](https://ads.tiktok.com/marketing_api/apps/).
   * @return appId
  **/
  @Schema(required = true, description = "The App ID applied by the developer. It can be found in the Basic Information section for you app under [My Apps](https://ads.tiktok.com/marketing_api/apps/).")
  public String getAppId() {
    return appId;
  }

  public void setAppId(String appId) {
    this.appId = appId;
  }

  public Oauth2AccessTokenBody authCode(String authCode) {
    this.authCode = authCode;
    return this;
  }

   /**
   * Authorization code provided once the callback is complete.
   * @return authCode
  **/
  @Schema(required = true, description = "Authorization code provided once the callback is complete.")
  public String getAuthCode() {
    return authCode;
  }

  public void setAuthCode(String authCode) {
    this.authCode = authCode;
  }

  public Oauth2AccessTokenBody secret(String secret) {
    this.secret = secret;
    return this;
  }

   /**
   * The private key of the developer&#x27;s App. It can be found in the Basic Information section for your app under [My Apps](https://ads.tiktok.com/marketing_api/apps/).
   * @return secret
  **/
  @Schema(required = true, description = "The private key of the developer's App. It can be found in the Basic Information section for your app under [My Apps](https://ads.tiktok.com/marketing_api/apps/).")
  public String getSecret() {
    return secret;
  }

  public void setSecret(String secret) {
    this.secret = secret;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Oauth2AccessTokenBody oauth2AccessTokenBody = (Oauth2AccessTokenBody) o;
    return Objects.equals(this.appId, oauth2AccessTokenBody.appId) &&
        Objects.equals(this.authCode, oauth2AccessTokenBody.authCode) &&
        Objects.equals(this.secret, oauth2AccessTokenBody.secret);
  }

  @Override
  public int hashCode() {
    return Objects.hash(appId, authCode, secret);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Oauth2AccessTokenBody {\n");
    
    sb.append("    appId: ").append(toIndentedString(appId)).append("\n");
    sb.append("    authCode: ").append(toIndentedString(authCode)).append("\n");
    sb.append("    secret: ").append(toIndentedString(secret)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
